cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -171,16 +203,24 @@ static void sc_mount_nvidia_driver_biarch(const char *rootfs_dir)
 {
 	// Bind mount a tmpfs on $rootfs_dir/var/lib/snapd/lib/gl
 	char buf[512] = { 0 };
+	char buf32[512] = { 0 };
 	sc_must_snprintf(buf, sizeof(buf), "%s%s", rootfs_dir,
 			 "/var/lib/snapd/lib/gl");
+	sc_must_snprintf(buf32, sizeof(buf32), "%s%s", rootfs_dir,
+			 "/var/lib/snapd/lib/gl/32");
 	const char *libgl_dir = buf;
 	debug("mounting tmpfs at %s", libgl_dir);
 	if (mount("none", libgl_dir, "tmpfs", MS_NODEV | MS_NOEXEC, NULL) != 0) {
 		die("cannot mount tmpfs at %s", libgl_dir);
 	};
+	if (mkdir(buf32, 00755) < 0) {
+		die("cannot create 32-bit subdirectory at %s", buf32);
+	}
 	// Populate libgl_dir with symlinks to libraries from hostfs
 	sc_populate_libgl_with_hostfs_symlinks(libgl_dir, nvidia_globs,
 					       nvidia_globs_len);
+	sc_populate_libgl_with_hostfs_symlinks(buf32, nvidia_globs32,
+					       nvidia_globs_len);
```

Comment:
IMO it would be sane to define `nvidia_globs_len32` too, unless we can guarantee the both sets of globs are of the same length.

Reply:
Agreed actually, I have a local vulkan change that sets that to len32